### PR TITLE
fix(test): wait enough time to Trigger Running Hook. Fixes: #12844

### LINKED ISSUE #12844: `TestTemplateLevelHooksWaitForTriggeredHook` fail sometimes (flakey test)
### Pre-requisites

- [X] I have double-checked my configuration
- [X] I can confirm the issue exists when I tested with `:latest`
- [X] I have searched existing issues and could not find a match for this bug
- [X] I'd like to contribute the fix myself (see [contributing guide](https://github.com/argoproj/argo-workflows/blob/main/docs/CONTRIBUTING.md))

### What happened/what did you expect to happen?

hook pod didn't created.

### Version

latest

### Paste a small workflow that reproduces the issue. We must be able to run the workflow; don't enter a workflows that uses private images.

```YAML
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  generateName: example-steps
spec:
  entrypoint: main
  templates:
    - name: main
      steps:
        - - name: job
            template: argosay
            hooks:
              running:
                expression: steps['job'].status == "Running"
                template: argosay-sleep-2seconds
              failed:
                expression: steps['job'].status == "Failed"
                template: argosay-sleep-2seconds

    - name: argosay
      container:
        image: argoproj/argosay:v2
        command: ["/bin/sh", "-c"]
        args: ["/bin/sleep 1; /argosay"]
    - name: argosay-sleep-2seconds
      container:
        image: argoproj/argosay:v2
        command: ["/bin/sh", "-c"]
        args: ["/bin/sleep 2; /argosay"]
```

Failure log:
```
✔ example-stepsjkrkr   Workflow  7s      
 └ ✔ example-stepsjkrkr Steps     7s      
 └ ✔ job                Pod       4s      
 └ ✔ [0]                StepGroup 7s      

Condition "to be Succeeded" met after 7s
Checking expectation example-stepsjkrkr
example-stepsjkrkr : Succeeded 
    hooks_test.go:506: 
        	Error Trace:	/home/runner/work/argo-workflows/argo-workflows/test/e2e/hooks_test.go:506
        	            				/home/runner/work/argo-workflows/argo-workflows/test/e2e/fixtures/then.go:69
        	            				/home/runner/work/argo-workflows/argo-workflows/test/e2e/fixtures/then.go:44
        	            				/home/runner/work/argo-workflows/argo-workflows/test/e2e/hooks_test.go:504
        	Error:      	Not equal: 
        	            	expected: "1/1"
        	            	actual  : "2/2"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,2 +1,2 @@
        	            	-(v1alpha1.Progress) (len=3) "1/1"
        	            	+(v1alpha1.Progress) (len=3) "2/2"
        	            	 
        	Test:       	TestHooksSuite/TestTemplateLevelHooksWaitForTriggeredHook
Checking expectation example-stepsjkrkr
example-stepsjkrkr : Succeeded 
Did not find node
    then.go:111: Did not find expected node
```

### Logs from the workflow controller

```text
kubectl logs -n argo deploy/workflow-controller | grep ${workflow}
```

### Logs from in your workflow's wait container

```text
kubectl logs -n argo -c wait -l workflows.argoproj.io/workflow=${workflow},workflow.argoproj.io/phase!=Succeeded
```

### PR DESCRIPTION
**Pull request by shuangkun**
sleep 1s is sometimes too short to trigger the running hook

Fixes #12844

### CODE REVIEW

File: test/e2e/hooks_test.go [hunk 1]
```diff
@@ -491,12 +491,12 @@ spec:
...
         image: argoproj/argosay:v2
         command: ["/bin/sh", "-c"]
-        args: ["/bin/sleep 1; /argosay"]
+        args: ["/bin/sleep 5; /argosay"]
     - name: argosay-sleep-2seconds
       container:
         image: argoproj/argosay:v2
         command: ["/bin/sh", "-c"]
-        args: ["/bin/sleep 2; /argosay"]
+        args: ["/bin/sleep 1; /argosay"]
```
Comment (tczhao) [inline test/e2e/hooks_test.go:499]: how about this, is reduced to 1s enough?
If so, could you update the step name to 1second

Comment (shuangkun) [reply test/e2e/hooks_test.go:499]: Oh, I feel like this sleep is of no use, so I turned it down. What works is the previous sleep. I forgot to change the name. Thanks

**Comment by shuangkun** [comment]
I saw there are many comment:
```	
	// TODO: Temporarily comment out this assertion since it's flaky:
	// 	  The running hook is occasionally not triggered. Possibly because the step finishes too quickly
	//	  while the controller did not get a chance to trigger this hook.
```
This one we can sleep longer. Because it only verifies running hooks, unlike others that verify other types of hooks.

File: test/e2e/hooks_test.go (hunk 1 above)
Comment (shuangkun) [reply test/e2e/hooks_test.go]: Returned to 2 because I saw many other similar ones using 2.

**Comment by tczhao** [comment]
Could you let us how you tested it to make sure it doesn't flake

**Comment by shuangkun** [comment]
> Could you let us how you tested it to make sure it doesn't flake

It is indeed not guaranteed, but now sleep 1s does often lead to a higher probability of failure of this test. In many other tests, the running hook check is skipped. If it is better for you to skip the task, you can skip it.

**Comment by tczhao** [comment]
I wonder if we can run https://argo-workflows.readthedocs.io/en/stable/running-locally/#running-e2e-tests-locally a few times to reproduce the original issue

**Comment by shuangkun** [comment]
> > Could you let us how you tested it to make sure it doesn't flake
> 
> It is indeed not guaranteed, but now sleep 1s does often lead to a higher probability of failure of this test. In many other tests, the running hook check is skipped. If it is better for you to skip the task, you can skip it.



> I wonder if we can run https://argo-workflows.readthedocs.io/en/stable/running-locally/#running-e2e-tests-locally a few times to reproduce the original issue

It is difficult to reproduce, but it may happen in the test after submitting the PR. Like TestStoppedWorkflow and TestTemplateExecutor, It is almost difficult to reproduce locally when running a test alone. It is easy to occur only after the PR is submitted and several people trigger the pipeline at the same time.

**Review by tczhao — APPROVED** [review]
I think we can keep monitor and see if this reduces flakiness

**Comment by shuangkun** [comment]
> I think we can keep monitor and see if this change reduces flakiness

Yes, I think so. Thanks

**Review by toyamagu-2021 — APPROVED** [review]
Thanks! We should observe this change resolves the issue.

> I wonder if we can run [argo-workflows.readthedocs.io/en/stable/running-locally/#running-e2e-tests-locally](https://argo-workflows.readthedocs.io/en/stable/running-locally/#running-e2e-tests-locally) a few times to reproduce the original issue

It might be hard to reproduce this issue by local testing, because this fakiness may be caused by poor spec of CI runner (2core, 7mem). 

Ref: https://github.com/argoproj/argo-workflows/pull/11534

@agilgur5 Thanks for mentioning me :)

**Comment by agilgur5** [comment]
> poor spec of CI runner (2core, 7mem).

This did get doubled [recently](https://github.blog/2024-01-17-github-hosted-runners-double-the-power-for-open-source/) (see also [Slack](https://cloud-native.slack.com/archives/C0510EUH90V/p1705542102584109?thread_ts=1705542102.584109&cid=C0510EUH90V) thread), it's now 4 vCPU & 16GB RAM. But 4 vCPU is _still_ likely less vCPU than most local dev setups

**Review by agilgur5 — APPROVED** [review]

**Pull request merged**